crates/driver/src/domain/competition/solution/fee.rs

```diff
@@ -75,46 +75,83 @@ impl Fulfillment {
 
     /// Computed protocol fee in surplus token.
     fn protocol_fee(&self, prices: ClearingPrices) -> Result<eth::TokenAmount, Error> {
-        // TODO: support multiple fee policies
-        if self.order().protocol_fees.len() > 1 {
-            return Err(Error::MultipleFeePolicies);
-        }
+        let mut current_order_price_limits = PriceLimits {
+            sell: self.order().sell.amount,
+            buy: self.order().buy.amount,
+        };
+        let mut fee = eth::TokenAmount::default();
+        self.order()
```

Comment:
style nit: Since this iterator is not very functional anyway (compute result outside the loop) having a simple `for` loop would help with readability here.